src/spotify_to_tidal/sync.py

```diff
@@ -222,7 +242,7 @@ def get_tracks_for_new_tidal_playlist(spotify_tracks: Sequence[t_spotify.Spotify
             seen_tracks.add(tidal_id)
     return output
 
-async def sync_playlist(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Session, spotify_playlist, tidal_playlist: tidalapi.Playlist | None, config):
+async def sync_playlist(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Session, spotify_playlist, tidal_playlist: tidalapi.Playlist | None, config, sync_favorites=False):
```

Comment:
I think the best way forward here is to generalize this function to something like below so that it can be used with both playlists and favourites:

```python
async def sync_tracks(spotify_tracks: Sequence[t_spotify.SpotifyTrack], old_tidal_tracks: Sequence[tidalapi.Track], config: Dict) -> Sequence[tidalapi.Track]:
```

and then have separate functions `sync_playlist` and `sync_favorites`